pkg/binding/selectors.go

```diff
@@ -139,6 +116,52 @@ func (c *Controller) updateResolutions(ctx context.Context, objIdentifier util.O
 	return nil
 }
 
+func (c *Controller) noteObjectForBindingPolicy(logger logr.Logger, objIdentifier util.ObjectIdentifier, objResourceVersion string,
+	matchedStatusCollectorsSet sets.Set[string], objBeingDeleted bool, bindingPolicyName string) error {
+	resolutionUpdated, err := c.bindingPolicyResolver.EnsureObjectIdentifierWithVersion(bindingPolicyName,
+		objIdentifier, objResourceVersion)
+	if err != nil {
+		if errorIsBindingPolicyResolutionNotFound(err) {
+			// this case can occur if a bindingpolicy resolution was deleted AFTER
+			// the BindingPolicyResolver::ResolutionExists call and BEFORE getting to the NoteObject function,
+			// which occurs if a bindingpolicy was deleted in this time-window.
+			logger.V(4).Info("skipped EnsureObjectIdentifierWithVersion for object because "+
+				"bindingpolicy was deleted", "objectIdentifier", objIdentifier,
+				"bindingpolicy", objResourceVersion)
+			return nil
+		}
+
+		return fmt.Errorf("failed to update resolution for bindingpolicy %s for object (identifier: %v): %v",
+			objResourceVersion, objIdentifier, err)
+	}
+
+	resolutionStatusCollectorsChanged, err := c.bindingPolicyResolver.SetStatusCollectorsForObject(bindingPolicyName,
+		objIdentifier, matchedStatusCollectorsSet)
+	if err != nil {
+		return fmt.Errorf("failed to set statuscollectors for object (identifier: %v): %v", objIdentifier,
```

Comment:
`errorIsBindingPolicyResolutionNotFound(err)` needs to be checked here too, it could have become true since the last check.
Better yet, merge EnsureObjectIdentifierWithVersion and SetStatusCollectorsForObject.

Reply:
done